### DISCUSSION TITLE
2021-10-19 weekly discussion

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#739] [Maintainer discussions]
### About
Each Tuesday, we start a discussion about Mathesar's progress and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

### Discussion topics
- [Progress](https://github.com/centerofci/mathesar/discussions/739#discussioncomment-1505162)
- [Design Process](https://github.com/centerofci/mathesar/discussions/739#discussioncomment-1505194)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/739#discussioncomment-1505196)
- [Weekly check-in](https://github.com/centerofci/mathesar/discussions/739#discussioncomment-1505198)

**Comment by kgodey** [comment]
## Progress

- We closed a feature milestone a couple of days ago! [05. Editable Tables](https://github.com/centerofci/mathesar/milestone/4) is now complete and we have the ability to add and edit records within Mathesar.
- We've continued to make progress on:
  - Backend and frontend work for [06. General Improvements #4](https://github.com/centerofci/mathesar/milestone/31)
  - Frontend work for [07. Working with Tables](https://github.com/centerofci/mathesar/milestone/8)
  - Backend work for [08. Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)
  - Design and product work for [09. Working with Views](https://github.com/centerofci/mathesar/milestone/13)
- We had a meeting about the latest Views milestone design, meeting notes are posted [on the wiki](https://wiki.mathesar.org/meeting-notes/october-2021) as usual.

**Comment by kgodey** [comment]
## Design Process

[09. Working with Views](https://github.com/centerofci/mathesar/milestone/13) involves many complex user interactions and problems to solve, including:
- conveying to the user the difference between tables and views and when to use them
- conveying data modeling concepts to the user, e.g. when to use one-to-many and many-to-many relationships
- different types of views
  - filters, sorts, and groups on a single tables
  - joining multiple tables together into a single view
  - views with computed columns based on aggregating, summarizing, or computing data from tables

We've been thinking about how to improve the design process for features like this and I'd like to open up a discussion about that. Please read through the Design Review channel on Matrix for some earlier conversation about this.

**Reply by kgodey** [reply]
Discussion on this topic continued on Matrix. Here's the process we settled on:

1. Kriti makes a design issue with initial requirements.
2. Ghislaine does some exploration through prototyping and Kriti provides feedback
3. Ghislaine updates the design issue with updated requirements/scenarios and creates additional placeholder issues as needed
4. The entire team reviews the updated issue and provides feedback on the issue before Ghislaine proceeds further with design.
5. Ghislaine finishes the prototype and posts the spec for review on GitHub Discussions.

**Comment by kgodey** [comment]
## Comms Assignee schedule

Here's the schedule for the next five weeks:

| Week | Person |
|-|-|
| Oct 18 to Oct 22 | @kgodey |
| Oct 25 to Oct 29 | @pavish |
| Nov 1 to Nov 5 | @seancolsen |
| Nov 8 to Nov 12 | @ghislaineguerin |
| Nov 15 to Nov 19 | @mathemancer |

**Comment by kgodey** [comment]
## Weekly check-in

@mathemancer, @ghislaineguerin, @pavish, @seancolsen, you are encouraged to respond. Others are also welcome to respond.

- What went well over the last week?
- What could have gone better over the last week?
- What are you planning to work on over the next week?
- Do you anticipate any blockers? Can anyone else on the team help?
- Anything else you'd like to add?

**Reply by kgodey** [reply]
> What went well over the last week?

I'm pretty excited about the editable tables milestone being completed. I also think our Views design meeting (although it went over time and all over the place) was a step in the right direction and helped identify several issues with our product planning that otherwise we might not have caught until implementation.

> What could have gone better over the last week?

I did not have any time to code, although I did do some speccing in #731. I've gone ahead and moved my backend issues back to the backlog since I think my time is currently best spent on product and planning work.

> What are you planning to work on over the next week?

- Mathesar 2022 budget planning
- Views product work
- Updating community and core team documentation on the wiki

> Do you anticipate any blockers? Can anyone else on the team help?

I don't anticipate any blockers.

> Anything else you'd like to add?

I'm looking forward to hiring a backend engineer so that our backend work moves faster and I don't have to feel as guilty about not having time for it. :)

**Reply by mathemancer** [reply]
> What went well over the last week?

I'm enjoying working with the back end candidates, we have some great ones.  I'm also happy to see outside contributors participating.

> What could have gone better over the last week?

I'm having trouble getting enough contiguous time for "deep work", which is keeping my velocity lower than I'd like.  This will probably be better as the new baby gets older.

> What are you planning to work on over the next week?

I'm working with back end candidates, and I want to finish up the range grouping.  This will be for more than just numeric types.

> Do you anticipate any blockers? Can anyone else on the team help?

I'll have to wait till the column display options work is done before doing the custom money type or percentile type work.

> Anything else you'd like to add?

Nothing at the moment.

**Reply by seancolsen** [reply]
> What went well over the last week?

- Feeling good about [adding support for nested Dropdown components](https://github.com/centerofci/mathesar/issues/732) which ended up being an interesting problem.
- Also feeling very energized to be working on open source software again. Seeing a new contributor [express interest in helping with a ticket I opened](https://github.com/centerofci/mathesar/issues/729#issuecomment-944695509), and [helping an established contributor solve a problem](https://github.com/centerofci/mathesar/pull/738#issuecomment-947123457) were delightful reminders that our community extends beyond the core team.

> What could have gone better over the last week?

- I'm still tweaking my notification settings across all our various communication channels so I've had a lot of distractions and not as much time for focused work as I'd like. But I think as I get used to the flow and culture here bit more I'll become more comfortable with taking longer to respond to some messages while I focus on work.
- Also eager to hear feedback from others about my performance and participation so far and ways I can work to optimize it.

> What are you planning to work on over the next week?

- Still taking this one day at a time until I get more deeply integrated and become a more autonomous.

> Do you anticipate any blockers? Can anyone else on the team help?

- Nope.

> Anything else you'd like to add?

- Thanks to everyone for welcoming me and helping me to get acquainted with the workflow here! This is a wonderful team which has created a very supportive environment for my first week!
- Especially excited about working more with Pavish because I can tell I'm going to learn a lot from him!

**Reply by ghislaineguerin** [reply]
> What went well over the last week?

- Feeling good with closing the old review discussions and starting new issues. It was important to address the different parts with more detail, the work feels more focused.
- Student project discussions were interesting, uncovering use cases for mathesar

> What could have gone better over the last week?

- Design review meeting could have been better organized

> What are you planning to work on over the next week?

- New issues for views / prototyping

> Do you anticipate any blockers? Can anyone else on the team help?

- None



> Anything else you'd like to add?

**Reply by pavish** [reply]
> What went well over the last week?

  I'm glad that editable tables is complete. I've also been focused more on frontend discussions and reviewing PRs.

> What could have gone better over the last week?

  I hoped to get the integration test environment set up, but I ran into a bunch of issues and having to read through a lot of documentation. I don't think I could have done anything different to speed it up though.

> What are you planning to work on over the next week?

  My plans are the same as last week. I plan to get the integ test env setup and write tests for features completed upto editable tables (I'm not sure how much I'll be able to complete). I also plan to spec out issues for next milestones and be more focused on reviewing PRs.

> Do you anticipate any blockers? Can anyone else on the team help?

  None, at the moment.

> Anything else you'd like to add?

  I've been feeling extra guilty about not completing features ever since I took up the refactoring, and now setting up the integration tests. I'm really glad @seancolsen is here and our feature completion is progressing steadily. I'm also very thankful for @mr-gabe49 for helping us out consistently!
